tests/import_test.go

```diff
@@ -89,6 +90,34 @@ var _ = Describe("[rfe_id:1115][crit:high][vendor:cnv-qe@redhat.com][level:compo
 		Entry("fail given a large physical size QCOW2 file", utils.LargePhysicalDiskQcow),
 	)
 
+	DescribeTable(`Should fail with "ImagePullFailed" reason if failed to pull image`, func(url string) {
+		dv := utils.NewDataVolumeWithRegistryImport("failed-to-pull-image", "10Gi", "docker://"+url)
+		if dv.Annotations == nil {
+			dv.Annotations = make(map[string]string)
+		}
+		dv.Annotations[controller.AnnImmediateBinding] = "true"
+
+		var err error
+		dv, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dv)
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = utils.WaitForPVC(f.K8sClient, dv.Namespace, dv.Name)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verify dImagePullFailed running condition")
+		runningCondition := &cdiv1.DataVolumeCondition{
+			Type:    cdiv1.DataVolumeRunning,
+			Status:  v1.ConditionFalse,
+			Message: common.ImagePullFailureText,
+			Reason:  "ImagePullFailed",
+		}
+		utils.WaitForConditions(f, dv.Name, f.Namespace.Name, controllerSkipPVCCompleteTimeout, assertionPollInterval, runningCondition)
+
+	},
+		Entry("insecure registry", "registry:5000/myimage:latest"),
```

Comment:
I wanted to catch to cases here that should be easy to reproduce: insecure registry and non exist image tag. But I don't think the test will fail outside of kvci, because it will still won't be able to pull the image, for other reasons (e.g. registry not found)

Reply:
yeah it fails on
```
transport.go:78] Could not create image reference: pinging container registry registry:5000: Get "https://registry:5000/v2/": dial tcp: lookup registry on 172.30.0.10:53: no such host
```
so should return your new err.

but how come this tests insecure registry? AFAIK the kubevirtci VM has this registry whitelisted

Reply:
There is a field in CDI CR for that. I'm not sure what is it setting in kvci, but locally this is what I'm getting.